### DISCUSSION TITLE
Please make the documentation for 'commit-ish' more explicit.

### DISCUSSION THREAD

**Discussion by pepasibble** [npm/feedback#476] [Website]
Hello,

In the docs on this page: https://docs.npmjs.com/cli/v7/configuring-npm/package-json there is a section which describes the "commit-ish" feature of specifying dependencies.

However, the description of 'commit-ish' and the examples given don't really match up.

The description of 'commit-ish' is:

> "If #commit-ish is provided, it will be used to clone exactly that commit."

Ok, so that lines up with this example:

```
"mocha": "mochajs/mocha#4727d357ea",
```

But then several other examples are given:

```
git+ssh://git@github.com:npm/cli.git#v1.0.27
```

```
"module": "user/repo#feature\/branch"
```

So it looks like 'commit-ish' can also be the name of a branch? Or maybe it can also be the name of a tag or release name in github?  Do you see how this isn't clear?

It would be great if the docs were more explicit about exactly what commit-ish can (and cannot) be.  For example, I would love to see documentation like this:

> "Commit-ish can be a commit hash, or the name of a branch, or the name of a tag."

Thanks!

**Comment by ljharb** [comment]
This is a git term - it can be anything that points to a commit. `HEAD`, `HEAD~1`, `branch`, `branch~3`, `branch^`, etc.

Perhaps just linking to https://git-scm.com/docs/gitglossary#Documentation/gitglossary.txt-aiddefcommit-ishacommit-ishalsocommittish would be useful?

**Reply by MylesBorins** [reply]
Please feel free to open an issue to improve the documentation on this
